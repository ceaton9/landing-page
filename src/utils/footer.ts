export const FOOTER = {
  FOOTER_ITEMS: [
    {
      TITLE: 'Email',
      LINKS: [
        {
          label: 'admin@grvstudio.id',
          url: 'mailto:admin@grvstudio.id',
        },
      ],
    },
    {
      TITLE: 'Home',
      LINKS: [
        {
          label: 'Product',
          url: '/',
        },
        {
          label: 'Project',
          url: '/',
        },
      ],
    },
    {
      TITLE: 'About',
      LINKS: [
        {
          label: 'Contact Us',
          url: '/',
        },
        {
          label: 'License',
          url: '/',
        },
        {
          label: 'Privacy Policy',
          url: '/',
        },
      ],
    },
    {
      TITLE: 'Address',
      LINKS: [
        {
          label: 'Bandung, West Java',
          url: '/',
        },
      ],
    },
  ],
}
