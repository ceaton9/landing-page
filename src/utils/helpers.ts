/**
 * @author Groove Studio
 * @copyright Â© All rights reserved. Groove Studio 2023
 */

import { join, split, toLower } from 'lodash'

export const toHTMLHypen = (word: string) =>
  toLower(join(split(word, ' '), '-'))
