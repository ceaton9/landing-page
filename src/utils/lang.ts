export const LANG = {
  HEADLINE: "Get Smart Solutions For Your Business with GROOVE.",
  MINI_HEADLINE: "BUSINESS BOOST",
  HEADLINE_DESC:
    "We're bringing digital creators, Solving problems with unique and creative system for your business",
  NAVBAR: {
    LOGO: "GRV.",
  },
};
