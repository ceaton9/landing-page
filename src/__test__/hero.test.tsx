/**
 * @author PT Docheck Bagi Indonesia
 * @copyright Â© All rights reserved. DoCheck 2022
 */

import React from "react";
import renderer from "react-test-renderer";

import Hero from "../components/hero";
import { LANG } from "../utils/lang";

describe("Hero", () => {
  const elemenToRender = (
    <Hero
      headline={LANG.HEADLINE}
      miniHeadline={LANG.MINI_HEADLINE}
      headlineDescription={LANG.HEADLINE_DESC}
    />
  );

  it("renders correctly", () => {
    const tree = renderer.create(elemenToRender).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders last word correctly", () => {
    const instance = renderer.create(elemenToRender).toTree();
    expect(instance?.props.headline.split(" ").pop()).toEqual("GROOVE.");
  });

  it("renders headline correctly", () => {
    const instance = renderer.create(elemenToRender).toTree();
    const lastsIndexOfSpace = instance?.props.headline.lastIndexOf(" ");

    expect(instance?.props.headline?.substring(0, lastsIndexOfSpace)).toEqual(
      "Get Smart Solutions For Your Business with"
    );
  });
});
