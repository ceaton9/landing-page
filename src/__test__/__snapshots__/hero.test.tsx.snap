// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<section
  className="flex px-24 pt-24"
>
  <section
    className="w-4/6 mr-36"
  >
    <p
      className="text-sm text-[#C4C4C4]"
    >
      BUSINESS BOOST
    </p>
    <h1
      className="text-white text-[56px] "
    >
      Get Smart Solutions For Your Business with 
      <span
        className="font-bold text-transparent bg-clip-text bg-gradient-to-r from-[#DD31D1] via-[#8343DA] to-[#3D84E7]"
      >
        GROOVE.
      </span>
    </h1>
    <p
      className="text-white opacity-70"
    >
      We're bringing digital creators, Solving problems with unique and creative system for your business
    </p>
  </section>
  <section
    className="w-3/6"
  >
    <div
      className="bg-teal-300 w-full h-full"
    />
  </section>
</section>
`;
